controllers/main.js

```diff
@@ -22,20 +22,42 @@ const getUrl = async (req, res) => {
     }
     res.json(url);
   } catch (error) {
-    res.status(HTTP_ERRORS.INTERNAL_SERVER_ERROR).json({ error: 'Something went wrong. Try Again' });
+    res
+      .status(HTTP_ERRORS.INTERNAL_SERVER_ERROR)
+      .json({ error: 'Something went wrong. Try Again' });
   }
 };
 
 const editUrl = async (req, res) => {
-  const { originalUrl, customName, shortUrl } = req.body;
+  const { originalUrl, customName } = req.body;
   const { id } = req.params;
   const filter = { _id: id };
-  const url = await Url.findOneAndUpdate(
-    filter,
-    { originalUrl, customName, shortUrl },
-    { new: true }
-  );
-  res.json(url);
+  let shortUrls;
+
+  try {
+    if (customName) {
+      const customNameWithDashes = customName.split(' ').join('-');
+      shortUrls = `${BASEURL}/${customNameWithDashes}`;
+    }
+
+    const url = await Url.findOneAndUpdate(
+      filter,
+      { customName, shortUrl: shortUrls },
```

Comment:
Why are you not updating the `originalUrl`?

Reply:
I think because, it was not supposed to be required and i felt like if i put it then it could mess up what i did

Reply:
Update it when it's available and don't update it when it's not sent. 😄

Reply:
done on 3a08efb